frame/support/procedural/src/benchmark.rs

```diff
@@ -145,16 +166,38 @@ struct BenchmarkDef {
 	setup_stmts: Vec<Stmt>,
 	call_def: BenchmarkCallDef,
 	verify_stmts: Vec<Stmt>,
+	return_type: Option<Type>, // `None` if `()` return type
 	extra: bool,
 	skip_meta: bool,
 }
 
+/// Parses the return type of a benchmark function definition
+fn parse_return_type(item_fn: &ItemFn) -> Result<Option<Type>> {
+	match &item_fn.sig.output {
+		ReturnType::Default => {
+			// return type is `()` which is permitted
+			// check for question mark ops as these aren't allowed here
+			let mut visitor = ExprTryVisitor::new();
+			visitor.visit_item_fn(item_fn);
+			if let Some(expr) = visitor.results.first() {
```

Comment:
I think its good to have the syntax clarification, but not sure if this check here actually works.  
It would still be possible to just `return Err("".into())` instead of using `?` and it would exclude `#[block]`s from using it.  
I dont think we can/have to explicitly check this, or?

Reply:
We could have a visitor that checks all return statements and applies a `static_assertions::assert_type_eq_all!` check to each one verifying that it is compatible with `BenchmarkResult`, and that wouldn't be too hard, but honestly now that `BenchmarkResult` is so short I am leaning towards just requiring `-> BenchmarkResult` on _all_ benchmark function defs, which would greatly simplify things and wouldn't be a very heavy lift. Thoughts?

Reply:
Either way would work though I think